src/coreclr/src/jit/lower.cpp

```diff
@@ -3231,8 +3295,24 @@ void Lowering::LowerRetStruct(GenTreeUnOp* ret)
         }
         break;
 
-        default:
+#if defined(TARGET_AMD64) || defined(TARGET_ARM64) || defined(TARGET_ARM)
+        case GT_CNS_DBL:
+            // Currently we are not promoting structs with a single float field.
+            // TODO: can improve `GT_CNS_DBL` handling for supported platforms, but
```

Comment:
Should probably be `TODO-CQ`,and you might add a reference to this from issue #4323, or vice-versa.